Lib/test/test_asyncio/test_base_events.py

```diff
@@ -231,6 +231,25 @@ def test_set_default_executor_error(self):
 
         self.assertIsNone(self.loop._default_executor)
 
+    def test_shutdown_default_executor_timeout(self):
+        class DummyExecutor(concurrent.futures.ThreadPoolExecutor):
+            def shutdown(self, wait=True, *, cancel_futures=False):
+                if wait:
+                    time.sleep(0.1)
+
+        self.loop._process_events = mock.Mock()
+        self.loop._write_to_self = mock.Mock()
+        executor = DummyExecutor()
+        self.loop.set_default_executor(executor)
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("default")
```

Comment:
I see. It seems that in tests, using `catch_warnings()` requires `simplefilter()`, otherwise the test fails when using `-Werror`. Is there any particular reason you're not using `assertWarns()` though? It takes care of this.

Reply:
No reason (besides me not knowing about that method, sorry). Updated!